pkg/util/workqueue/default_rate_limiters.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workqueue
+
+import (
+	"sync"
+	"time"
+
+	"github.com/juju/ratelimit"
+)
+
+type RateLimiter interface {
```

Comment:
> We already have some rate limiting stuff in util/? Is it not reusable here? I'd prefer not to add a rate limiter to the workqueue package.

It's a slight different interface for the queuer to allow it automatically backoff per-item as well as per-queue.  There is an adapter for handling the bucket rate limiter we use as a godep so that you can reliably rate limit both the controller and the items to prevent excessive burst utilization by a single item.